### PR TITLE
Remove Elasticsearch

### PR DESCRIPTION
**Pull request by dblenkus**

### CODE REVIEW

File: tests/urls.py [hunk 1]
```diff
@@ -3,22 +3,23 @@
...
-search_router = search_routers.SearchRouter(trailing_slash=False)  # pylint: disable=invalid-name
+search_router = SearchRouter(trailing_slash=False)
 search_router.register(r'kb/feature/search', FeatureSearchViewSet, 'kb_feature_search')
 search_router.register(r'kb/feature/autocomplete', FeatureAutocompleteViewSet, 'kb_feature_autocomplete')
 search_router.register(r'kb/mapping/search', MappingSearchViewSet, 'kb_mapping_search')
 
-urlpatterns = [  # pylint: disable=invalid-name
+urlpatterns = [
     path('api-auth/', include('rest_framework.urls', namespace='rest_framework')),
     # XXX: Temporary fix to work with Resolwe 2.0.0, which requires 'resolwe-api' namespace to be available when
```
Comment (gregorjerse) [inline tests/urls.py:24]: Maybe rename temporary fix to permanent?

File: tests/settings.py [hunk 2]
```diff
@@ -275,15 +269,5 @@
             'handlers': ['file'],
             'level': 'DEBUG',
         },
-        'elasticsearch': {
-            'handlers': ['file'],
-            'level': 'WARNING',
-            'propagate': False,
-        },
-        'urllib3': {
```
Comment (gregorjerse) [inline tests/settings.py:283]: This part has probably nothing to do with elastic search removal?

File: resolwe_bio/kb/models.py [hunk 3]
```diff
@@ -101,20 +117,35 @@ class Mapping(models.Model):
...
-            ["target_db", "target_id", "target_species"],
+        indexes = [
+            models.Index(
+                name="idx_feature_source_target",
+                fields=[
+                    "source_db",
+                    "source_id",
+                    "source_species",
+                    "target_db",
+                    "target_species",
```
Comment (gregorjerse) [inline resolwe_bio/kb/models.py:142]: Should target_species also be added to the first index?

File: resolwe_bio/kb/migrations/0009_full_text_search.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        ),
+        migrations.AddIndex(
+            model_name="mapping",
+            index=models.Index(
+                fields=[
+                    "source_db",
+                    "source_id",
+                    "source_species",
+                    "target_db",
+                    "target_species",
```
Comment (gregorjerse) [inline resolwe_bio/kb/migrations/0009_full_text_search.py:66]: Same question as bellow: should target_species be included in this index?

File: resolwe_bio/kb/filters.py [hunk 5]
```diff
@@ -5,11 +5,114 @@
...
+
+
+class FullTextFilter(filters.BaseCSVFilter):
+    """Filter for full-text search."""
+
+    def filter(self, queryset, value):
+        """Filter field by text-search vector.
+
+        Values, given in csv format are joined with the OR operator.
+
```
Comment (gregorjerse) [inline resolwe_bio/kb/filters.py:27]: Maybe consider renaming second argument from value to values?

File: resolwe_bio/kb/filters.py [hunk 6]
```diff
@@ -5,11 +5,114 @@
...
+        Filtering by more than 135 values raises recursion error, so we have
+        to do it in chunks.
+        """
```
Comment (gregorjerse) [inline resolwe_bio/kb/filters.py:30]: Just out of curiosity: is the recursion error raised by the database or Django ORM?

File: resolwe_bio/kb/filters.py [hunk 7]
```diff
@@ -5,11 +5,114 @@
...
+        if not value:
+            return queryset
+
+        value = list(map(lambda v: SearchQuery(v, config="simple"), value))
```
Comment (gregorjerse) [inline resolwe_bio/kb/filters.py:34]: Is not the list comprehension nicer is this situation?

`value = [SearchQuery(v, config="simple") for v in value]`

File: resolwe_bio/kb/filters.py [hunk 8]
```diff
@@ -5,11 +5,114 @@
...
+
+        step = 100
+        result_qs = queryset.none()
+        for i in range(0, len(value), step):
+            query = reduce(SearchQuery.__or__, value[i : i + step])
```
Comment (gregorjerse) [inline resolwe_bio/kb/filters.py:39]: Nice :+1:

File: resolwe_bio/kb/filters.py (hunk 5 above)
Comment (dblenkus) [reply resolwe_bio/kb/filters.py:27]: Done.

File: resolwe_bio/kb/filters.py (hunk 6 above)
Comment (dblenkus) [reply resolwe_bio/kb/filters.py:30]: Max recursion depth is exceded in python when Django tries to resolwe the filer.

File: resolwe_bio/kb/models.py (hunk 3 above)
Comment (dblenkus) [reply resolwe_bio/kb/models.py:142]: What do you mean by "first index"?

**Pull request merged**